packages/mobile/e2e/src/usecases/NewAccountPhoneVerification.js

```diff
@@ -0,0 +1,252 @@
+import { TWILIO_ACCOUNT_SID, TWILIO_AUTH_TOKEN, VERIFICATION_PHONE_NUMBER } from '@env'
+import { dismissBanners } from '../utils/banners'
+import { EXAMPLE_NAME, EXAMPLE_PHONE_NUMBER } from '../utils/consts'
+import { checkKomenci } from '../utils/komenci'
+import { checkBalance, receiveSms } from '../utils/twilio'
+import { enterPinUi, setUrlDenyList, sleep, waitForElementId } from '../utils/utils'
+
+const jestExpect = require('expect')
+const examplePhoneNumber = VERIFICATION_PHONE_NUMBER || EXAMPLE_PHONE_NUMBER
+
+export default NewAccountPhoneVerification = () => {
+  beforeEach(async () => {
+    await device.launchApp({
+      delete: true,
+      permissions: { notifications: 'YES', contacts: 'YES' },
+    })
+    // Enable url deny list
+    await setUrlDenyList()
+
+    // Dismiss banners for firebase warning
+    await dismissBanners()
+
+    // Proceed through education screens
+    for (let i = 0; i < 3; i++) {
+      await element(by.id('Education/progressButton')).tap()
+    }
+
+    // Create new account
+    await element(by.id('CreateAccountButton')).tap()
+
+    // Accept terms
+    await element(by.id('scrollView')).scrollTo('bottom')
+    await expect(element(by.id('AcceptTermsButton'))).toBeVisible()
+    await element(by.id('AcceptTermsButton')).tap()
+
+    // Set name
+    await element(by.id('NameEntry')).replaceText(EXAMPLE_NAME)
+    await element(by.id('NameAndPictureContinueButton')).tap()
+
+    // Set and verify pin
+    await enterPinUi()
+    await enterPinUi()
+
+    // Set phone number
+    await expect(element(by.id('PhoneNumberField'))).toBeVisible()
+    await element(by.id('PhoneNumberField')).replaceText(examplePhoneNumber)
+    await element(by.id('PhoneNumberField')).tapReturnKey()
+  })
+
+  // Uninstall after to remove verification
+  afterAll(async () => {
+    device.uninstallApp()
+  })
+
+  // Check Twilio balance at start
+  beforeAll(async () => {
+    try {
+      await checkBalance()
+    } catch {}
+  })
+
+  // Check that Twilio SID, Auth Token and Verification Phone Number are defined
+  // Check that Twilio balance is enought to complete tests
+  if (
+    TWILIO_ACCOUNT_SID &&
+    TWILIO_AUTH_TOKEN &&
+    VERIFICATION_PHONE_NUMBER &&
+    (async () => await checkBalance())
```

Comment:
This is just a function, it's not actually waiting for the `checkBalance` return. I think you can just remove the `async () => `

Reply:
Changed the logic for this in https://github.com/valora-inc/wallet/pull/534/commits/e2fd0d47293a2784f6d92b1a54a9b9fce7c2982d. As conditional testing isn't possible with an async request. See [jest/issues/7245](https://github.com/facebook/jest/issues/7245).